timbuctoo-instancev4/src/main/java/nl/knaw/huygens/timbuctoo/rdf/Database.java

```diff
@@ -218,6 +220,21 @@ public RelationType findOrCreateRelationType(Node predicate) {
 
     if (relationTypeT.hasNext()) {
       return new RelationType(relationTypeT.next());
+    } else {
+      final GraphTraversal<Vertex, Vertex> inverseTypes = graphWrapper
```

Comment:
I'd use the rdfuri index for this, this is really slow and I guess has to happen a lot.